ConfluentSchemaRegistry IntegrationTest prints huge logs

https://github.com/apache/pinot/runs/7890686433?check_suite_focus=true
```
2022-08-18T02:12:33.6547446Z [INFO] Running org.apache.pinot.integration.tests.KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest
...
2022-08-18T02:14:26.6174836Z 02:14:25.310 ERROR [KafkaConfluentSchemaRegistryAvroMessageDecoder] [mytable__0__0__20220818T0214Z] Caught exception while decoding row in topic KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest, discarding row
2022-08-18T02:14:26.6175975Z org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
2022-08-18T02:14:26.6176584Z Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!
2022-08-18T02:14:26.6177510Z 02:14:25.867 ERROR [KafkaConfluentSchemaRegistryAvroMessageDecoder] [mytable__1__0__20220818T0214Z] Caught exception while decoding row in topic KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest, discarding row
2022-08-18T02:14:26.6178539Z org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
2022-08-18T02:14:26.6179139Z Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!
...
```
going on for 120,000 lines.

```
2022-08-18T02:14:26.6175975Z org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
2022-08-18T02:14:26.6176584Z Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!
```
This happens when trying to use the confluent schema registry to deserialize a message that was written using the standard (non-schemaregistry) avro serializer. I'll take a quick look at this.

--- a/pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest.java
+++ b/pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest.java
@@ -75,6 +75,7 @@ public class KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegr
   private static final List<String> UPDATED_INVERTED_INDEX_COLUMNS = Collections.singletonList("DivActualElapsedTime");
   private static final long RANDOM_SEED = System.currentTimeMillis();
   private static final Random RANDOM = new Random(RANDOM_SEED);
+  private static final int NUM_INVALID_RECORDS = 5;
 
   private final boolean _isDirectAlloc = RANDOM.nextBoolean();
   private final boolean _isConsumerDirConfigured = RANDOM.nextBoolean();
@@ -129,29 +130,39 @@ public class KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegr
         "io.confluent.kafka.serializers.KafkaAvroSerializer");
     Producer<byte[], GenericRecord> avroProducer = new KafkaProducer<>(avroProducerProps);
 
+    // this producer produces intentionally malformatted records so that
+    // we can test the behavior when consuming such records
     Properties nonAvroProducerProps = new Properties();
     nonAvroProducerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:" + getKafkaPort());
     nonAvroProducerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
         "org.apache.kafka.common.serialization.ByteArraySerializer");
     nonAvroProducerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
         "org.apache.kafka.common.serialization.ByteArraySerializer");
-    Producer<byte[], byte[]> nonAvroProducer = new KafkaProducer<>(nonAvroProducerProps);
+    Producer<byte[], byte[]> invalidDataProducer = new KafkaProducer<>(nonAvroProducerProps);
 
     if (injectTombstones()) {
       // publish lots of tombstones to livelock the consumer if it can't handle this properly
       for (int i = 0; i < 1000; i++) {
         // publish a tombstone first
-        nonAvroProducer.send(
+        avroProducer.send(
             new ProducerRecord<>(getKafkaTopic(), Longs.toByteArray(System.currentTimeMillis()), null));
       }
     }
+
     for (File avroFile : avroFiles) {
+      int numInvalidRecords = 0;
       try (DataFileStream<GenericRecord> reader = AvroUtils.getAvroReader(avroFile)) {
         for (GenericRecord genericRecord : reader) {
           byte[] keyBytes = (getPartitionColumn() == null) ? Longs.toByteArray(System.currentTimeMillis())
               : (genericRecord.get(getPartitionColumn())).toString().getBytes();
-          // Ignore getKafkaMessageHeader()
-          nonAvroProducer.send(new ProducerRecord<>(getKafkaTopic(), keyBytes, "Rubbish".getBytes(UTF_8)));
+
+          if (numInvalidRecords < NUM_INVALID_RECORDS) {
+            // send a few rubbish records to validate that the consumer will skip over non-avro records, but
+            // don't spam them every time as it causes log spam
+            invalidDataProducer.send(new ProducerRecord<>(getKafkaTopic(), keyBytes, "Rubbish".getBytes(UTF_8)));
+            numInvalidRecords++;
+          }
+
           avroProducer.send(new ProducerRecord<>(getKafkaTopic(), keyBytes, genericRecord));
         }
       }

--- a/pinot-integration-tests/src/test/resources/log4j2.xml
+++ b/pinot-integration-tests/src/test/resources/log4j2.xml
@@ -26,11 +26,6 @@
     </Console>
   </Appenders>
   <Loggers>
-    <!--Turn off the logger for KafkaConfluentSchemaRegistryAvroMessageDecoder because we intentionally inject
-     tombstones in KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest which can flood the log
-     -->
-    <Logger name="org.apache.pinot.plugin.inputformat.avro.confluent.KafkaConfluentSchemaRegistryAvroMessageDecoder"
-            level="off" additivity="false"/>
     <Logger name="org.apache.pinot" level="warn" additivity="false">
       <AppenderRef ref="console"/>
     </Logger>

